internal/pikamon/spawn/pokemon.go

```diff
@@ -0,0 +1,68 @@
+package spawn
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/dgraph-io/ristretto"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type pokemonSpawner struct {
+	chance         float64
+	channelCache   *ristretto.Cache
+	debounceWindow time.Duration
+}
+
+func newPokemonSpawner(chance float64, debounceWindow time.Duration) (*pokemonSpawner, error) {
+	cache, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: 1e2,
+		MaxCost:     1e2,
+		BufferItems: 64,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create channel cache")
+	}
+	return &pokemonSpawner{
+		chance:         chance,
+		channelCache:   cache,
+		debounceWindow: debounceWindow,
+	}, nil
+}
+
+func (p *pokemonSpawner) spawn(s *discordgo.Session, m *discordgo.MessageCreate) bool {
+	// check if channel id is still in cache, if it is
+	// we are still in the debounce window
+	_, exists := p.channelCache.Get(m.ChannelID)
+	if exists {
+		return false
+	}
+
+	// randomly determine if a pokemon is spawned
+	if rand.Float64() > p.chance {
+		return false
+	}
+
+	// spawn a pokemon!
+	pokemonID := rand.Intn(890) + 1
```

Comment:
assuming this means there are currently only 890 Pokemon accessible through `https://pokeres.bastionbot.org/images/pokemon/`?

Reply:
there are only 890 Pokemon in existence haha.

Reply:
I actually didn't know that lol